app/scripts/lib/createMetaRPCHandler.js

```diff
@@ -20,6 +52,19 @@ const createMetaRPCHandler = (api, outStream) => {
     let error;
     try {
       result = await api[data.method](...data.params);
+
+      if (
```

Comment:
Do we want to persist state upon failure as well? Some error cases still involve important state changes, and an error response is still a response.

Reply:
yep makes sense to me

Reply:
Done [here](https://github.com/MetaMask/metamask-extension/pull/15978/commits/d00d9577d808a346cfa5a03f4463602d144edf6e)